Fix #9

KasperFranz opened the pull request:
This PR fixes the reported issue

KasperFranz committed:
Fix #9

matkoniecz commented on test_wikidata_structure.py:
```diff
@@ -38,6 +38,7 @@ def is_not_a_behavior(self, type_id):
         self.is_not_a_specific_error_class(type_id, 'a human behavior')
 
     def is_not_a_specific_error_class(self, type_id, expected_error_class):
+        wikimedia_connection.set_cache_location(osm_handling_config.get_wikimedia_connection_cache_location())
```
is it needed?

KasperFranz replied:
I had issues running a single test in isolation.

I have pushed another fix where it is just using the setup_module https://docs.pytest.org/en/7.1.x/how-to/xunit_setup.html

matkoniecz replied:
oh, so you were using IDE that can trigger single test, right?

KasperFranz replied:
Yes I am using PyCharm.

I have also executed all the tests as part of this change and no new failures

KasperFranz committed:
Move to setup_module

matkoniecz commented:
Curiously, tests fail for me with

```
======================================================================
ERROR: test_well_formed_nonexisting_wikidata (test_wikimedia_link_issue_reporter.Tests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mateusz/Documents/install_moje/OSM_software/wikibrain_py_package_published/test_wikimedia_link_issue_reporter.py", line 173, in test_well_formed_nonexisting_wikidata
    self.assertNotEqual(None, self.detector().critical_structural_issue_report('node', {'wikidata': 'Q812843783738234723482347238272487927', 'wikipedia': 'en:Oslo'}))
  File "/home/mateusz/Documents/install_moje/OSM_software/wikibrain_py_package_published/wikibrain/wikimedia_link_issue_reporter.py", line 316, in critical_structural_issue_report
    something_reportable = self.check_is_wikidata_page_existing(key, tags.get(key))
  File "/home/mateusz/Documents/install_moje/OSM_software/wikibrain_py_package_published/wikibrain/wikimedia_link_issue_reporter.py", line 520, in check_is_wikidata_page_existing
    wikidata = wikimedia_connection.get_data_from_wikidata_by_id(present_wikidata_id)
  File "/home/mateusz/.local/lib/python3.8/site-packages/wikimedia_connection/wikimedia_connection.py", line 561, in get_data_from_wikidata_by_id
    if it_is_necessary_to_reload_wikidata_by_id_files(wikidata_id) or forced_refresh:
  File "/home/mateusz/.local/lib/python3.8/site-packages/wikimedia_connection/wikimedia_connection.py", line 592, in it_is_necessary_to_reload_wikidata_by_id_files
    content_filename = get_filename_with_wikidata_entity_by_id(wikidata_id)
  File "/home/mateusz/.local/lib/python3.8/site-packages/wikimedia_connection/wikimedia_connection.py", line 440, in get_filename_with_wikidata_entity_by_id
    return os.path.join(cache_location(), cache_folder_name(), wikidata_language_placeholder(), get_form_of_link_usable_as_filename(id) + ".wikidata_entity.txt")
  File "/home/mateusz/.local/lib/python3.8/site-packages/wikimedia_connection/wikimedia_connection.py", line 430, in cache_location
    assert cache_location_store != None, "wikimedia_connection.set_cache_location must be called before that point"
NameError: name 'cache_location_store' is not defined
```

etc. I will try to look is it broken on my side.

KasperFranz commented:
That is interesting, leave it with me to verify a few things

KasperFranz committed:
Add setupModule as well (required by https://docs.python.org/3/library/unittest.html#setupmodule-and-teardownmodule) where pytest is using setup_module

KasperFranz commented:
I have just pushed a fix for that, it seems like the library I am using is using a new style

matkoniecz committed:
automatically flush cache on failing brand test

matkoniecz committed:
more tests for failing cases

matkoniecz committed:
drop not needed comment

matkoniecz committed:
do as linter asked

matkoniecz committed:
add hint how it can be fixed

matkoniecz commented:
I am still getting `NameError: name 'cache_location_store' is not defined`

matkoniecz committed:
finish rename

now malformed secondary wikipedia tags are reported from two places what is kind of substandard, yes

KasperFranz commented:
With the last changes pushed by you, do you still get the issue? 

if you are what commands are you running?

KasperFranz committed:
Move to a test helper to simplify the setup, this also adds a new tests_cache directory only used for the unit tests

KasperFranz committed:
Fix wrong function

KasperFranz commented:
I have pushed in a test helper and moved to use a local cache directory for the test (this makes it possible to clear the test cache without having to clear the global cache)
